src/Microsoft.AspNetCore.Mvc.DataAnnotations/MvcDataAnnotationsLocalizationConfigureCompatibilityOptions.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Mvc.Infrastructure;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.Mvc.DataAnnotations
+{
+    internal class MvcDataAnnotationsLocalizationConfigureCompatibilityOptions : ConfigureCompatibilityOptions<MvcDataAnnotationsLocalizationOptions>
+    {
+        public MvcDataAnnotationsLocalizationConfigureCompatibilityOptions(
+            ILoggerFactory loggerFactory,
+            IOptions<MvcCompatibilityOptions> compatibilityOptions)
+            :base(loggerFactory, compatibilityOptions)
+        {
+        }
+
+        protected override IReadOnlyDictionary<string, object> DefaultValues
+        {
+            get
+            {
+                var values = new Dictionary<string, object>();
+
+                if (Version > CompatibilityVersion.Version_2_2)
```

Comment:
Notice that this is `Version > CompatibilityVersion.Version_2_2` and not `Version >= CompatibilityVersion.Version_2_2`. Meaning no behavior change by default for 2.2.

Reply:
This is different than the way we do in other places: https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Formatters.Json/MvcJsonOptionsConfigureCompatibilityOptions.cs#L26

Some scenarios to consider:
- An application was written for 2.0/2.1 and now they are updating their packages to 2.2 because they want to try out new features and expect the existing code to work as it is. (with `>= Version_2_2`, this scenario should just work as their code wouldn't be having the `.SetCompatibilityVersion(Microsoft.AspNetCore.Mvc.CompatibilityVersion.Version_2_2)`)
- An application was written for 2.0/2.1 and now they are updating their packages to 2.2 because they want to try out new features + want to use the 2.2 compat switch expecting the behavior changes they want (they expect the behavior change, so its ok to have the `>=` instead of `>`)
- 2.2 templates would by default come with 2.2 compat switch i.e `.SetCompatibilityVersion(Microsoft.AspNetCore.Mvc.CompatibilityVersion.Version_2_2)` (so its expected to have the behavior change, so its ok to have the `>=`)

Reply:
That's what I thought and why I called it out, but it wasn't what I understood from my conversation with @mkArtakMSFT and @danroth27. I'll change it to `>=` unless one of them specifically objects here.